Marlin/src/feature/mixing.h

```diff
@@ -27,23 +27,30 @@
   #define MIXER_ACCU_SIGNED
   typedef uint8_t mixer_color_t;
   typedef int8_t mixer_accu_t;
+  #define COLOR_A_MASK 0x80
+  #define COLOR_MASK 0x7F
 #else
   typedef uint_fast16_t mixer_color_t;
   typedef uint_fast16_t mixer_accu_t;
+  #define COLOR_A_MASK 0x8000
+  #define COLOR_MASK 0x7FFF
 #endif
 
-#define COLOR_A_MASK _BV(sizeof(mixer_color_t) * 8 - 1) // 0x80 or 0x8000
-#define COLOR_MASK (COLOR_A_MASK - 1)                   // 0x7F or 0x7FFF
-
 #ifndef MIXING_VIRTUAL_TOOLS
   #define MIXING_VIRTUAL_TOOLS 1
 #endif
 
 #ifdef RETRACT_SYNC_MIXING
   #define NR_MIXING_VIRTUAL_TOOLS (MIXING_VIRTUAL_TOOLS + 1)
   #define MIXER_AUTORETRACT_TOOL MIXING_VIRTUAL_TOOLS
+  #if NR_MIXING_VIRTUAL_TOOLS > 255
+    #error("MIXING_VIRTUAL_TOOLS must be <= 254!");
```

Comment:
Is this `#error` message incorrect or is the `#if` statement incorrect?

Reply:
Neither nor. 
If RETRACT_SYNC_MIXING is defined **NR_** MIXING_VIRTUAL_TOOLS is MIXING_VIRTUAL_TOOLS + 1. 
 the last tool is the retract tool.
Because the user has no direct influence on **NR_** MIXING_VIRTUAL_TOOLS but only on MIXING_VIRTUAL_TOOLS the error talks about MIXING_VIRTUAL_TOOLS

The limit has to exist because the used index is a uint8_t and a user already tried to configure 256 v-tools. A conditional typedef would be possible but even more confusing.